docs/_layouts/default.html

```diff
@@ -90,6 +90,32 @@
   <div id="fb-root"></div>
 
   <script>
+    (function() {
+      var selector = '.inner-content h2, .inner-content h3, .inner-content h4';
+      var elements = document.querySelectorAll(selector);
+      for (var i = 0; i < elements.length; i++) {
+        var elementID;
+
+        if (elements[i].hasAttribute('id')) {
+          elementID = elements[i].getAttribute('id');
+        } else {
+          var textMethod = document.body.textContent ? "textContent" : "innerText";
+          var roughText = elements[i][textMethod];
+
+          // Regex rule for making the title URL-friendly.
+          tidyText = roughText.toLowerCase()
+                              .replace(/\s+/g, '-')
+                              .replace(/[^A-Za-z0-9\-_.]/g, '');
+
+          elements[i].setAttribute('id', tidyText);
```

Comment:
The anchors do work. I did some testing across the various desktop browsers where I added anchors at various points in the loading process. I found that they all hop to IDs added after the DOM is fully loaded, though they differ in how much later in the process they will successfully jump to the anchor. In case you're curious, here's where each of the browsers fared on the test (the browser name is listed where they last successfully jumped to the anchor).

* **On DOM Ready (DOMContentLoaded)**
  * Safari
  * Chrome
  * Opera
* **On Page Fully Loaded (window.onload)**
  * Firefox
  * IE
* **After Arbitrary setTimeout**
  * - none -

In short, we're safe in adding the IDs where we have, at least for all the browsers I tested.